### PR TITLE
Add travis build entry for liquid haskell checks

### PR DESCRIPTION
**Pull request by SooryaN**

### CODE REVIEW

**Comment by piyush-kurur** [comment]
Since you sqashed the changes, I am not sure if you are done with the update. Also interestingly this commit is only building against Appveyor and not travis which is where our interest lies. Not sure what is happening

**Comment by SooryaN** [comment]
Looks like it works now. The issue was that it wasn't able to parse the  `.travis.yml` file earlier. Missing quotes around the ppa for the apt sources.

**Comment by piyush-kurur** [comment]
There are couple of things.

- [] Rebase to  the current master
- [] Even when liquid haskell fail the build is not failing.

Of course liquid haskell should be in the allowed failures section. But then the failure should give a red cross instead of the green tick. My guess is that the liquid haskell executables's exit status is _not_ failure
even when the checks fail.

**Comment by SooryaN** [comment]
How do you propose I do the liquid failing check? I could pipe the output to a text file and grep for failure.
Would that be an acceptable solution?

**Comment by piyush-kurur** [comment]
On Thu, Jul 05, 2018 at 07:38:16AM -0700, J.M. Soorya Narayan wrote:
> How do you propose I do the liquid failing check? I could pipe the output to a text file and grep for failure.
> Would that be an acceptable solution?

I mean first confirm the exit status of the liquid haskell
executable. If it is not then it even makes sense to raise a ticket on
the liquid haskell repository.

**Comment by SooryaN** [comment]
Hey, sorry about the delay. I was traveling. 

So the exit status was always 0 because `find -exec` always returns that. I changed it to use `xargs`. It works locally but I'm not sure why it isn't reflected in the travis build.

File: scripts/run-lh.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+pwd
+echo "-------------------------------------------------"
+find . -name '*.hs' -print0 | xargs -n 1 -0 -I{} sh -c "echo ---- {}; $LIQUID {}"
+cd ..
+pwd
+echo "-------------------------------------------------"
+cd raaz-implementation
+pwd
+echo "-------------------------------------------------"
+find . -name '*.hs' -print0 | xargs -n 1 -0 -I{} sh -c "echo ---- {}; $LIQUID {}"
```
Comment (piyush-kurur) [inline scripts/run-lh.sh]: `xargs` or the find version I guess runs liquid haskell checks one at a time on each file. I am not sure if it is the correct thing to do. Shouldn't liquidhaskell get all the relevant files as command like arguments at one go as some of the files have dependencies ?

On in other words is something morally equivalent to
 
```bash
liquidhaskell `find . -name '*.hs'`
``` 
better ?

Comment (SooryaN) [reply scripts/run-lh.sh]: One issue with this is that LH complains when the same module name is found in multiple files. In this case, the Raaz.Entropy module.

As far as the dependencies are concerned, LH is able to identify them as long as there's a cabal file in the directory.

Comment (piyush-kurur) [reply scripts/run-lh.sh]: The Entropy module should be ignored. I do not think there is anything there that is of interest to LH

**Pull request merged**